File: backend/bird-pantanal-photo-gallery/src/main/java/com/projects/bird_pantanal_photo_gallery/dto/BirdDTO.java
package com.projects.bird_pantanal_photo_gallery.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record BirdDTO(
        @NotNull(message = "name must not be null") 
        @NotBlank(message = "name must not be blank") 
        String name,

        @NotNull(message = "scientific name must not be null") 
        @NotBlank(message = "scientific name must not be blank") 
        String scientificName,

        @NotNull(message = "description must not be null") 
        @NotBlank(message = "description must not be blank") 
        String description,

        @NotNull(message = "predominant color must not be null") 
        @NotBlank(message = "predominant color must not be blank") 
        String predominantColor) {
}

File: frontend/bird-pantanal-photo-gallery/src/app/dto/bird.dto.ts

export class BirdDTO {
    name: string;
    scientificName: string;
    description: string;
    predominantColor: string;
  
    constructor(
      name: string,
      scientificName: string,
      description: string,
      predominantColor: string
    ) {
      this.name = name;
      this.scientificName = scientificName;
      this.description = description;
      this.predominantColor = predominantColor;
    }
  }
  